pallets/clusters/src/lib.rs

```diff
@@ -142,16 +152,17 @@ pub mod pallet {
 		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
 		pub fn create_cluster(origin: OriginFor<T>, name: Vec<u8>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			ensure!(!name.len().is_zero(), Error::<T>::SystematicFailure);
+			ensure!(!name.len().is_zero(), Error::<T>::InvalidInputs);
 
 			let cluster_hash = blake2_256(&name);
-
+			// Check that the cluster does not exist already
 			ensure!(!<Clusters<T>>::contains_key(&cluster_hash), Error::<T>::ClusterExists);
```

Comment:
I'm actually unsure about this, we might want clusters with the same name.
I think the cluster can be a 128 bit hash, and a random hash is fine (pseudo, mix e.g. (extrinsic index+origin+name))

Reply:
We want to add Cluster ID to aliases types too

Reply:
A random hash brings in the problem of how to retrieve it from other functions later on. For example, when the owner of the Cluster wants to create a new Role and add it to an existing Cluster (which is empty at the beginning, no Members and no Roles), he needs a way to retrieve the hash of that Cluster. But if there are multiple clusters with the same name and same owner, there is no way to recreate the hash of the exact Cluster he wants to add the Role to.
Also Protos and Fragments have a hash that can be generated from the data they contain, it's not random.

So, I am thinking to change the returned value of the extrinsic create_cluster to `DispatchResultWithPostInfo` so that the random cluster_id generated can be returned to caller and be used with the other extrinsics (e.g.` create_role`, `add_member`, etc...). It would be a first time in our codebase, though.

Reply:
Good idea! Updated it to be [random](https://github.com/fragcolor-xyz/clamor/blob/8276f14d8e8d6eb42a1fcc163adbc348619888ed/pallets/clusters/src/lib.rs#L177) and also created a [cluster_id](https://github.com/fragcolor-xyz/clamor/blob/8276f14d8e8d6eb42a1fcc163adbc348619888ed/pallets/clusters/src/lib.rs#L58) field that is used by the other extrinsic and will be used for the pallet-aliases too.